src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -46,6 +46,12 @@ def _get_dumper_function(destination: Union[Parameter, Artifact]) -> Callable:
     if isinstance(destination, Parameter) and destination.dumps is not None:
         return destination.dumps
 
+    if isinstance(destination, Artifact):
+        if destination.dumps is not None:
+            return destination.dumps
+        if destination.dumpb is not None:
+            return destination.dumpb
```

Comment:
Note we can simplify this
```suggestion
    if destination.dumps is not None:
        return destination.dumps
    if isinstance(destination, Artifact) and destination.dumpb is not None:
        return destination.dumpb
```